comm/src/main/scala/coop/rchain/comm/network.scala

```diff
@@ -0,0 +1,116 @@
+package coop.rchain.comm
+
+import java.net.SocketTimeoutException
+import scala.util.{Failure, Success, Try}
+import scala.collection.concurrent
+import scala.concurrent.{Await, Promise}
+import scala.concurrent.duration.{Duration, MILLISECONDS}
+import coop.rchain.kademlia.PeerTable
+
+/**
+  * Implements the lower levels of the network protocol.
+  */
+case class UnicastNetwork(id: NodeIdentifier, endpoint: Endpoint) extends ProtocolHandler {
+
+  case class PendingKey(remote: Seq[Byte], timestamp: Long, seq: Long)
+
+  val pending =
+    new concurrent.TrieMap[PendingKey, Promise[Try[ProtocolMessage]]]
+
+  val local = new ProtocolNode(id, endpoint, this)
+  val comm = UnicastComm(local)
+  val table = PeerTable(local)
+
+  private val receiver = new Thread {
+    override def run =
+      while (true) {
+        comm.recv match {
+          case Success(res) =>
+            for {
+              msg <- ProtocolMessage.parse(res)
+            } dispatch(msg)
+          case Failure(ex: SocketTimeoutException) => ()
+          case Failure(ex)                         => ex.printStackTrace
+        }
+      }
+  }.start
```

Comment:
Do you really need the `receiver` variable? The `Thread#start` method returns `Unit`.
Also, I found another way to write this code. Scala 2.12 has SAM support, so you can write something like this:
```val task: Runnable = () => { //body }
val receiver = new Thread(task).start()```